crates/ibc/src/core/ics24_host/identifier.rs

```diff
@@ -177,6 +178,48 @@ impl From<String> for ChainId {
     }
 }
 
+#[cfg_attr(
+    feature = "parity-scale-codec",
+    derive(
+        parity_scale_codec::Encode,
+        parity_scale_codec::Decode,
+        scale_info::TypeInfo
+    )
+)]
+#[cfg_attr(
+    feature = "borsh",
+    derive(borsh::BorshSerialize, borsh::BorshDeserialize)
+)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+/// Type of the client, depending on the specific consensus algorithm.
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct ClientType(String);
+
+impl ClientType {
+    /// Constructs a new instance without performing any validation primarily for use in testing.
+    pub(crate) fn new_unchecked(s: String) -> Self {
```

Comment:
the `_unchecked` postfix is typically reserved for unsafe functions ([example](https://doc.rust-lang.org/std/pin/struct.Pin.html#method.get_unchecked_mut)). I suggest we instead put this as `From<String>` instead

Reply:
[1396ad6](https://github.com/cosmos/ibc-rs/pull/639/commits/1396ad6c82c454444114461f7f55c4509a4727a5)